src/parser/ParsedQuery.h

```diff
@@ -358,6 +358,10 @@ class ParsedQuery {
 struct GraphPatternOperation {
   struct BasicGraphPattern {
     vector<SparqlTriple> _triples;
+
+    void appendTriples(const BasicGraphPattern& pattern) {
+      ad_utility::appendVector(_triples, pattern._triples);
```

Comment:
```suggestion
    void appendTriples(BasicGraphPattern pattern) {
      ad_utility::appendVector(_triples, std::move(pattern._triples));
```
(Only one overload, one Copy, if we pass in something not-moved, zero copies if we move in, one more move than necessary, but that is very cheap). You can remember this pattern if you always need to copy the arguments of a function.